src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4240,13 +4240,13 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     </Copy>
 
-    <!-- If this project produces reference assemblies *and* copied (possibly transitive)
+    <!-- If this project produces reference assemblies *or* copied (possibly transitive)
          references on this build, subsequent builds of projects that depend on it must
          not be considered up to date, so touch this marker file that is considered an
          input to projects that reference this one. -->
     <Touch Files="$(CopyUpToDateMarker)"
            AlwaysCreate="true"
-           Condition="'$(ProduceReferenceAssembly)' == 'true' and '@(ReferencesCopiedInThisBuild)' != ''" />
+           Condition="'$(ProduceReferenceAssembly)' == 'true' or '@(ReferencesCopiedInThisBuild)' != ''" />
```

Comment:
In VS the ClassLibrary1 marker wouldn't be updated on a noop build because MSBuild wouldn't be invoked. On the command line the markers would be updated for every project on every build, which is unnecessary I/O, but nothing should look at them so it's probably not the end of the world.

The design we were working with dealt with this scenario by adding the implementation assemblies to the list of FUTD inputs. Can you elaborate on why you prefer this design?